### ISSUE TITLE
Actions class throws an error on mouseMove method when using RemoteWebDriver

### ISSUE THREAD

**Issue by SpyderWeiss** [SeleniumHQ/selenium#4015]
## Meta -
OS:  OSX 10.12.4 & Linux 6.9
Software used:  Selenium 3.4.0, selenium-server-standalone-3.4.0.jar, Intellij Ultimate 2017.1.2, Java 8 update 131 
Browser:  Firefox
Browser Version:  53.0.2 (64-bit)

## Expected Behavior
Element has mouse moved to it.

## Actual Behavior
Actions class throws an error on mouseMove method located in RemoteMouse line 89.  Note that this works fine with the WebDriver class, but not the RemoteWebDriver class (I've included examples of how we are building our driver for both).

## Steps to reproduce
Start up a Selenium grid with the standalone server:
java -Dwebdriver.gecko.driver=~/Stash/automation/src/test/Drivers/Mac\ OS\ X/ckodriver -Xmn2048m -Xms4096M -Xmx10240m -Xss8192k -jar ~/Downloads/Selenium/selenium-server-standalone-3.4.0.jar -role hub -hubConfig ~/Downloads/Selenium/hubconfig.json

Start up a Selenium grid node:
java -Dwebdriver.gecko.driver=~/Stash/automation/src/test/Drivers/Mac\ OS\ X/geckodriver -jar ~/Downloads/Selenium/selenium-server-standalone-3.4.0.jar -role node -nodeConfig ~/Downloads/Selenium/nodeconfig.json

(I can attach my hubconfig and nodeconfig if that's relevant)

Launch a remote web browser (use case GRID, I've included case FF in my code as well to show how we are launching our local browser that works fine with the Actions class:

private WebDriver webDriver = null;

    public Browser()
    {
        //Set log4j property so warning message doesn't appear in console
        System.setProperty("org.apache.commons.logging.Log", "org.apache.commons.logging.impl.Jdk14Logger");
        Dimension dimension = new Dimension(1400, 1080);
        String driverDir = "src/test/Drivers/" + System.getProperty("os.name");
        long startLaunch = System.nanoTime();
        DesiredCapabilities capabilities;
        String browserName = null;
        switch (driver)
        {
            case FF:
                System.setProperty("webdriver.gecko.driver", String.format("%s/geckodriver", driverDir));
                ProfilesIni allProfiles = new ProfilesIni();
                FirefoxProfile profile = allProfiles.getProfile("Automation");
                profile.setPreference("browser.startup.homepage", "about:blank");
                profile.setEnableNativeEvents(true);
                profile.setAcceptUntrustedCertificates(true);
                profile.setPreference("security.mixed_content.block_active_content", false);
                profile.setPreference("plugins.click_to_play", false);
	            profile.setPreference("media.autoplay.enabled", false);
                try
                {
                    webDriver = new FirefoxDriver(profile);
                    browserName = "Firefox";
                    resize(dimension);
                }
                catch (WebDriverException we)
                {
                    // when FF browser gets 'stuck' (in parallel execution mode), you can help the test continue by killing the smallest memory FF
                    // process  [windows only]
                    String checkError = we.getMessage();
                    //  if message contains text like "could not bind to port xxxx", kill it so browser can restart
                    if (checkError.contains("locking port"))
                    {
                        if (Config.debug)
                        {
                            Util.log("Intervene to free up locked FF port");
                        }
                        // this only works for Windows processes  (in cmd window, type TASKKILL /? for details)
                        try
                        {

                            Runtime.getRuntime().exec("taskkill /F /IM firefox.exe /FI \"MEMUSAGE lt 100,000K\"");
                            Runtime.getRuntime().runFinalization();
                        }
                        catch (IOException e)
                        {
                            if (Config.debug)
                            {
                                Util.log("Could not fix firefox lockup, trying Linux kill code.");
                            }
                            e.printStackTrace();
                            //adding this to kill linux processes
                            try
                            {

                                Runtime.getRuntime().exec("killall firefox");
                                Runtime.getRuntime().runFinalization();
                            }
                            catch (IOException e2)
                            {
                                e2.printStackTrace();
                                throw we;
                            }
                        }
                    }

                }
                catch (Exception ex)
                {
                    if (Config.debug)
                    {
                        Util.log("Connect to FF again");
                    }
                    webDriver = new FirefoxDriver(profile);
                    browserName = "Firefox";
                    resize(dimension);
                }
                break;
            case GRID:
                capabilities = DesiredCapabilities.firefox();
                capabilities.setCapability("marionette", true);
                capabilities.setCapability("plugins.click_to_play", false);
		        capabilities.setCapability("media.autoplay.enabled", false);
                setUpGrid(capabilities);
                browserName = "Firefox";
                resize(dimension);
                break;
            default:
                throw new WebDriverException("Unrecognized driver specification");
        }
        //Safari does not support timeouts, it will fail if this if statement is not here
        if (webDriver != null && !webDriver.toString().contains("safari"))
        {
            webDriver.manage().timeouts().pageLoadTimeout(360, TimeUnit.SECONDS);
            webDriver.manage().timeouts().setScriptTimeout(360, TimeUnit.SECONDS);
            webDriver.manage().timeouts().implicitlyWait(0, TimeUnit.MILLISECONDS);
        }
        long endLaunch = System.nanoTime();

        Calendar cal = Calendar.getInstance();
        DateFormat df = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.MEDIUM);

        Util.log("%s browser launched in %f seconds at %s", browserName, Util.convertNSToSeconds(endLaunch - startLaunch), df.format(cal.getTime()));
        BaseTestCases.browserCount(true);
    }

    private void setUpGrid(Capabilities capabilities)
    {
        int maxAttempts = 20;
        int tryCount = 0;
        String hostName;
        int port = 4444;
        do
        {
            try
            {
                RemoteWebDriver remoteWebDriver = new RemoteWebDriver(new URL("http://localhost:4444/wd/hub"), capabilities);
                //Log the IP Address and name of the machine you are running the test on
                try
                {
                    CloseableHttpClient client = HttpClients.createDefault();
                    HttpGet httpGet = new HttpGet("http://" + hostName + ":" + port + "/grid/api/testsession?session=" + remoteWebDriver.getSessionId());
                    CloseableHttpResponse response = client.execute(httpGet);
                    JSONObject object = new JSONObject(EntityUtils.toString(response.getEntity()));
                    String proxyID = object.get("proxyId").toString();
                    String portNumber = proxyID.split("//")[1].split(":")[1];
                    portNumber = portNumber.replaceAll("44", "");
                    Util.log("Test running on IP Address: " + proxyID.split("//")[1].split(":")[0]
                            + " / Machine name: ussl-qasel" + portNumber.replaceFirst("44", ""));
                    webDriver = remoteWebDriver;

                }
                catch (JSONException | IOException e)
                {
                    Util.log("Exception " + e + " occured.");
                }
                break;
            }
            catch (MalformedURLException malUrl)
            {
                malUrl.getStackTrace();
                break;
            }
            catch (WebDriverException webDvrEx)
            {
                // Check if it is an "Unable to bind to locking port 7054 within 45000 ms" error
                if (webDvrEx.getMessage().contains("locking port"))
                {
                    tryCount++;
                    Util.log("Making attempt #%d to get driver after receiving \"Unable to bind to locking port 7054\" WebDriver error!", tryCount + 1);
                    if (webDriver != null)
                    {
                        webDriver.quit();
                        //Util.waitMilliseconds(5000, "Wait for driver to quit");
                    }
                    //else
                    //{
                    //    Util.waitMilliseconds(5000, "Wait before attempting to get another WebDriver");
                    //}
                }
                //without an else here the tryCount never increases and you get caught in an infinite loop
                else
                {
                    tryCount++;
                    Util.log(webDvrEx.getMessage() + "Retrying webDriver connection, attempt number: " + tryCount);
                }
            }
            Util.waitMilliseconds(5000, "Wait before attempting to get another WebDriver");
        } while (tryCount < maxAttempts);

        if (tryCount >= maxAttempts)
        {
            if (webDriver == null)
            {
                throw new RuntimeException(String.format("Unable to get webDriver after %d attempts!", maxAttempts));
            }
        }
    }
	
    /**
     * Method to resize the browser window
     *
     * @param dimension Size to make the browser window
     */
    void resize(Dimension dimension)
    {
        Util.log("Resize window %s", dimension.toString());
        webDriver.manage().window().setSize(dimension);
    }

And then go to a webpage and attempt to move the mouse to an element on that page (the focus() method in the stacktrace at the bottom calls just this code):
new Actions(webDriver).moveToElement(webElement).click().perform();

You should get the error:
org.openqa.selenium.UnsupportedCommandException: mouseMoveTo
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'mikesMacbookPro', ip: 'XX.XX.XXX.XXX', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.4', java.version: '1.8.0_131'
Driver info: driver.version: RemoteWebDriver
Command duration or timeout: 19 milliseconds

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:215)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:167)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:671)
	at org.openqa.selenium.remote.RemoteExecuteMethod.execute(RemoteExecuteMethod.java:35)
	at org.openqa.selenium.remote.RemoteMouse.mouseMove(RemoteMouse.java:89)
	at org.openqa.selenium.interactions.MoveMouseAction.perform(MoveMouseAction.java:43)
	at org.openqa.selenium.interactions.CompositeAction.perform(CompositeAction.java:36)
	at org.openqa.selenium.interactions.Actions$BuiltAction.perform(Actions.java:634)
	at org.openqa.selenium.interactions.Actions.perform(Actions.java:593)
	at Integration.Framework.Element.focus(Element.java:1288)
	at Integration.Framework.TextField.setText(TextField.java:65)
	at Integration.Framework.TextField.setText(TextField.java:54)
	at Integration.EAD.Hub.Pages.HubLoginPage.setUserName(HubLoginPage.java:108)
	at Integration.EAD.Hub.Pages.HubLoginPage.login(HubLoginPage.java:146)
	at Integration.EAD.Hub.Pages.HubLoginPage.login(HubLoginPage.java:84)
	at Integration.EAD.Hub.TestCases.HubLoginAndLogout.test(HubLoginAndLogout.java:28)
	at Integration.EAD.Hub.TestCases.HubLoginAndLogout.setup(HubLoginAndLogout.java:106)
	at Integration.EAD.Hub.TestCases.HubLoginAndLogout.World_Preferred_HubLoginAndLogout(HubLoginAndLogout.java:175)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:75)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:14)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.openqa.selenium.UnsupportedCommandException: mouseMoveTo
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'mikesMacbookPro', ip: 'XX.XX.XXX.XXX', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.4', java.version: '1.8.0_131'
Driver info: driver.version: RemoteWebDriver
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'mikesMacbookPro', ip: 'XX.XX.XXX.XXX', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12.4', java.version: '1.8.0_131'
Driver info: driver.version: RemoteWebDriver
	at org.openqa.selenium.remote.http.AbstractHttpCommandCodec.encode(AbstractHttpCommandCodec.java:220)
	at org.openqa.selenium.remote.http.AbstractHttpCommandCodec.encode(AbstractHttpCommandCodec.java:118)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:637)
	at org.openqa.selenium.remote.RemoteExecuteMethod.execute(RemoteExecuteMethod.java:35)
	at org.openqa.selenium.remote.RemoteMouse.mouseMove(RemoteMouse.java:89)
	at org.openqa.selenium.support.events.internal.EventFiringMouse.mouseMove(EventFiringMouse.java:58)
	at org.openqa.selenium.remote.server.handler.interactions.MouseMoveToLocation.call(MouseMoveToLocation.java:59)
	at org.openqa.selenium.remote.server.handler.interactions.MouseMoveToLocation.call(MouseMoveToLocation.java:32)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
	... 3 more

**Comment by lmtierney** [comment]
Duplicate of #3693
